test/convert_to_json.cpp

```diff
@@ -0,0 +1,60 @@
+#include <migraphx/convert_to_json.hpp>
+#include <test.hpp>
+
+TEST_CASE(key_num)
+{
+    std::string str = "{abc:{key:1}}";
+    auto jstr       = migraphx::convert_to_json(str);
+    EXPECT(jstr == "{\"abc\":{\"key\":1}}");
+}
+
+TEST_CASE(key_null)
+{
+    std::string str = "{abc:{key:null}}";
+    auto jstr       = migraphx::convert_to_json(str);
+    EXPECT(jstr == "{\"abc\":{\"key\":null}}");
+}
+
+TEST_CASE(key_nan)
+{
+    std::string str = "{abc:{key:nan}}";
+    auto jstr       = migraphx::convert_to_json(str);
+    EXPECT(jstr == "{\"abc\":{\"key\":nan}}");
+}
+
+TEST_CASE(quote_key_num)
+{
+    std::string str = R"({"abc":{"key":1}})";
+    auto jstr       = migraphx::convert_to_json(str);
+    EXPECT(jstr == "{\"abc\":{\"key\":1}}");
+}
+
+TEST_CASE(quote_with_space_key_num)
+{
+    std::string str = R"({"abc key":{"key":1}})";
+    auto jstr       = migraphx::convert_to_json(str);
+    EXPECT(jstr == "{\"abc key\":{\"key\":1}}");
+}
+
+TEST_CASE(key_value_num_space)
+{
+    std::string str = "{abc    :    {    key    :    1}}";
+    auto jstr       = migraphx::convert_to_json(str);
+    EXPECT(jstr == "{\"abc\"    :    {    \"key\"    :    1}}");
+}
+
+TEST_CASE(key_value_str)
+{
+    std::string str = "{abc : {key : value}}";
+    auto jstr       = migraphx::convert_to_json(str);
+    EXPECT(jstr == "{\"abc\" : {\"key\" : \"value\"}}");
+}
+
+TEST_CASE(key_space_value)
+{
+    std::string str = "{abc    : [key, value]}";
+    auto jstr       = migraphx::convert_to_json(str);
+    EXPECT(jstr == "{\"abc\"    : [\"key\", \"value\"]}");
+}
```

Comment:
There is some more test cases that probably should be added:

* `{ key: 1 }`
* `{ key: -1 }`
* `{ key: 1.0 }`
* `{ key: -1.0 }`
* `{ key: 1e-10 }`
* `{ key: 1E+10 }`
* `{ key: nan }`
* `{ key: true }`
* `{ key: false }`
* `{ key: null }`

Quotes should be around the `key` but not the value.

Reply:
added a few more unit tests